File: Programming_Task_04_ShaliniSengar/C_Files/File_Handling_Challenge.c
#include <stdio.h>

int main() {
    char name[50], branch[50];
    int rollNo, marks;

    // Input student details
    printf("Enter Name: ");
    scanf(" %[^\n]", name);

    printf("Enter Roll Number: ");
    scanf("%d", &rollNo);

    printf("Enter Branch: ");
    scanf("%s", branch);

    printf("Enter Marks: ");
    scanf("%d", &marks);

    // Write data to file
    FILE *fp = fopen("student_data.txt", "w");

    fprintf(fp, "Name: %s\n", name);
    fprintf(fp, "Roll No: %d\n", rollNo);
    fprintf(fp, "Branch: %s\n", branch);
    fprintf(fp, "Marks: %d\n", marks);

    fclose(fp);

    printf("\nStudent Record Saved Successfully\n");

    // Read and display file contents
    fp = fopen("student_data.txt", "r");

    char ch;
    printf("\nReading File...\n\n");

    while ((ch = fgetc(fp)) != EOF) {
        printf("%c", ch);
    }

    fclose(fp);

    return 0;
}

File: Programming_Task_04_ShaliniSengar/Python_Files/File_Handling_Challenge.py
# Input student details
name = input("Enter Name: ")
roll_no = int(input("Enter Roll Number: "))
branch = input("Enter Branch: ")
marks = int(input("Enter Marks: "))

# Write data to file
with open("student_data.txt", "w") as file:
    file.write(f"Name: {name}\n")
    file.write(f"Roll No: {roll_no}\n")
    file.write(f"Branch: {branch}\n")
    file.write(f"Marks: {marks}\n")

print("\nStudent Record Saved Successfully")

# Read and display file contents
print("\nReading File...\n")

with open("student_data.txt", "r") as file:
    content = file.read()
    print(content)
